### PR TITLE
[O11y][SpringBoot] Add custom field for _id and update dashboard

### PR DESCRIPTION
**Pull request by rajvi-elastic**
- Enhancement

## What does this PR do?
- Create one custom field `spring_boot.audit_events.document_id` and copy the value of `_id` field into `spring_boot.audit_events.document_id` .
- Replace `_id` field with `spring_boot.audit_events.document_id` in d.ashboard.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Screenshots

3892080395-image

## Related Issues
- #5871
- #5221 

> **_Note_**  When user will upgrade to the newer version, `spring_boot.audit_events.document_id` field will be added only to the new incoming documents. The data that is already present in indices will not get updated. Hence, the Audit Events [Logs Spring Boot] visualization will show documents which have `spring_boot.audit_events.document_id` field present (documents that have been ingested after upgrading to the latest version).

> By default, the Include empty row property is disabled till 8.1.0. But in later version of Kibana the empty rows are displayed if this property is not configured and because of that the visualization shows multiple empty rows. To avoid this and to keep the panel functionality intact we have migrated to 8.2.0.

Behavior of panel in 8.2.0 and later version without  configuration:
image

Behavior of panel in 8.2.0 and later version with configuration:
image

### CODE REVIEW

File: packages/spring_boot/_dev/build/docs/README.md [hunk 1]
```diff
@@ -26,6 +26,99 @@ In order to ingest data from Spring Boot:
...
 
+### Troubleshooting
+
+If **[Spring Boot] Audit Events panel** does not display older documents after upgrading to ``0.9.0`` or later versions, then this issue can be solved by reindexing the ``audit_events`` data stream's indices.
+
+To reindex the data, the following steps must be performed.
+
+1. Stop the data stream by going to `Integrations -> Spring Boot -> Integration policies` open the configuration of Spring Boot and disable the `Spring Boot Audit Events metrics` toggle to reindex ``audit_events`` data stream and save the integration.
+
+2. Perform the following steps in the Dev tools.
```
Comment (agithomas) [inline packages/spring_boot/_dev/build/docs/README.md:37]: For steps 2 , 4, 7 mention what is the goal of the step before the phrase -  `Perform the following steps in Dev Tools.`.

**Review by agithomas — APPROVED** [review]
LGTM!

Approving based on the discussion with the product team that - it is Ok not to show _id field values for old records , reindexing can be recommended in case the user wish to see the old information following the change (adoption of document_id) as the new field to store the values presently stored under _id.

@rajvi-elastic , please complete the peer review additionally

cc @SubhrataK

**Review by milan-elastic — APPROVED** [review]

**Comment by rajvi-elastic** [comment]
@agithomas, We have created issue (https://github.com/elastic/kibana/issues/155344) to add support for meta fields. Should we wait for the Kibana team response or should we merge this PR?

**Comment by agithomas** [comment]
> @agithomas, We have created issue ([elastic/kibana#155344](https://github.com/elastic/kibana/issues/155344)) to add support for meta fields. Should we wait for the Kibana team response or should we merge this PR?

@lalit-satapathy , the above mentioned issue is created based on your request. Should this be a blocker ?

**Pull request merged**